### ISSUE TITLE
Feature request: re-usable components

### ISSUE THREAD

**Issue by jhenin** [Colvars/colvars#232]
In a first stage, these may be cvc objects that are thin wrappers around other cvcs and only implement getting the values, not applying forces.

**Comment by HanatoK** [comment]
Just curious about this feature. Does this mean we can do something like compute RMSD of dihedrals or other CVs without using custom functions or tcl scripts?

**Comment by jhenin** [comment]
For that I think you'll still need custom functions or scripts. This feature is useful when several variables are needed that are different combinations of the same components. Then they'll only need to be computed once. In a first "read-only" stage, it will allow doing analysis on individual components while using them in a separate colvar for biasing.

**Comment by giacomofiorin** [comment]
@HanatoK To compute a RMSD of a set of dihedral there is a `dihedralPC` component, which assumes input in the format used by CARMA: https://utopia.duth.gr/glykos/carma.html and uses `dihedral` components.  You can probably make a similar class with very small effort, should the scripted or custom function be too slow.

**Comment by giacomofiorin** [comment]
Looks like this needs #254 done before moving ahead. Assign back to me if needed.

**Comment by jhenin** [comment]
Hi @giacomofiorin This feature is getting more useful with additions by @HanatoK (PR coming soon). Another possible pathway is to make a colvar depend on another colvar, but this seems more lightweight in terms of architecture changes.

**Comment by giacomofiorin** [comment]
Good to know, I look forward to the PR.

I also agree that it's better to keep dependencies between components, since we have already a few ad-hoc cases and it is the components that take up the bulk of the computation.

**Comment by HanatoK** [comment]
Hi @giacomofiorin,
I don't know too much about `colvardeps`. Is that possible to treat the dependencies between components just the same as `colvarbias` and the components? If that can be implemented, then we can refer a CVC in different definition of `colvar {}` blocks to avoid redundant computations.

**Comment by giacomofiorin** [comment]
Yes, what you ask is possible in theory but not implemented yet.

**Comment by jhenin** [comment]
The current dependency system would allow a CVC to have several parent colvars, just the same way a colvar can have several parent biases.

**Comment by giacomofiorin** [comment]
> The current dependency system would allow a CVC to have several parent colvars, just the same way a colvar can have several parent biases.

But the loop over CVCs in the SMP version does not take that into account, and could easily end up invoking the same CVCs on multiple threads at the same time.

It's not a big change to work around that, but it needs to be done.
